src/openzaak/utils/permissions.py

```diff
@@ -138,13 +145,67 @@ def has_object_permission(self, request: Request, view, obj) -> bool:
         if not self.permission_fields:
             return request.jwt_auth.has_auth(scopes_required, component)
 
-        main_resource = self.get_main_resource()
+        main_resource = self.get_main_resource(self.main_resource)
 
         if view.__class__ is main_resource:
             main_object = obj
         else:
             main_object = self.get_main_object(obj, view.permission_main_object)
 
-        main_object_data = self.format_data(main_object, request)
-        fields = self.get_fields(main_object_data)
+        main_object_data = self.format_data(main_object, request, main_resource)
+        fields = self.get_fields(main_object_data, self.permission_fields)
         return request.jwt_auth.has_auth(scopes_required, component, **fields)
+
+
+class MultipleObjectsAuthRequired(AuthRequired):
+    permission_fields: dict
+    main_resources: dict
+
+    def get_field_viewset(self, request, viewset, action):
+        field_viewset = import_string(viewset)()
+        field_viewset.request = request
+        field_viewset.format_kwarg = "json"
```

Comment:
I don't think I've seen this before, why is this needed?

Reply:
get_serializer which was called in validate_create needs request and format_kwarg set, changed that to also expect the serializer_class as an argument so that the field viewset only needs the action.